webapp/packages/core-settings/src/createSettingsAliasResolver.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { ISyncExecutor, SyncExecutor } from '@cloudbeaver/core-executor';
+import type { schema } from '@cloudbeaver/core-utils';
+
+import type { ISettingChangeData, ISettingsSource } from './ISettingsSource';
+import type { SettingsProvider } from './SettingsProvider';
+
+const DEPRECATED_SETTINGS = new Set();
+
+type SettingsMapping<TTarget> = Partial<{
+  [key in keyof TTarget]: string;
+}>;
+
+export function createSettingsAliasResolver<TTarget extends schema.SomeZodObject>(
+  source: ISettingsSource,
+  target: SettingsProvider<TTarget>,
+  mappings: SettingsMapping<schema.infer<TTarget>>,
+): ISettingsSource {
+  type targetSchema = schema.infer<TTarget>;
+  const reversed = Object.fromEntries(Object.entries(mappings).map(a => a.reverse()));
```

Comment:
Sound like a helper function for me

Lets move it to helpers and do something like this. The map is optional, we dont need it right now


```suggestion
  function reverseObject<Key extends string, Value extends string>(obj: Record<Key, Value> | Map<Key, Value>): Record<Value, Key>{
    const reversedObj: Record<Value, Key> = {} as Record<Value, Key>;
    for (const key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
            reversedObj[obj[key]] = key;
            continue;
        }

        throw new Error('Object is not a Record');
    }
    return reversedObj;
}

function reverseMap<Key, Value>(map: Map<Key, Value>): Map<Value, Key> {
    const reversedMap = new Map<Value, Key>();
    for (let [key, value] of map) {
        reversedMap.set(value, key);
    }
    return reversedMap;
}
```

Reply:
replaced with underscore implementation